client/src/client-only-routes/ShowProjectLinks.js

```diff
@@ -0,0 +1,174 @@
+
+  const handleSolutionModalHide = () => setSolutionState(initSolutionState);
+
+  const getProjectSolution = (projectId, projectTitle) => {
+    const {
+      user: { completedChallenges }
+    } = props;
+
+    const completedProject = find(
+      completedChallenges,
+      ({ id }) => projectId === id
+    );
+
+    if (!completedProject) {
+      return null;
+    }
+
+    const { solution, githubLink, files } = completedProject;
+    const onClickHandler = () =>
+      setSolutionState({
+        projectTitle,
+        files,
+        solution,
+        isOpen: true
+      });
+
+    if (files && files.length) {
+      return (
+        <button
+          onClick={onClickHandler}
+          className='project-link-button-override'
+        >
+          solution
+        </button>
+      );
+    }
+    if (githubLink) {
+      return (
+        <>
+          <a href={solution} rel='noopener noreferrer' target='_blank'>
+            solution
+          </a>
+          ,{' '}
+          <a href={githubLink} rel='noopener noreferrer' target='_blank'>
+            source
+          </a>
+        </>
+      );
+    }
+    if (maybeUrlRE.test(solution)) {
+      return (
+        <a
+          block={'true'}
+          className='btn-invert'
+          href={solution}
+          rel='noopener noreferrer'
+          target='_blank'
+        >
+          solution
+        </a>
+      );
+    }
+    return (
+      <button className='project-link-button-override' onClick={onClickHandler}>
+        solution
+      </button>
+    );
+  };
+
+  const renderCertification = certName => (
+    <ul key={certName}>{renderProjectsFor(certName)}</ul>
+  );
+
+  const renderProjectsFor = certName => {
+    return projectMap[certName].map(({ link, title, id }) => (
+      <li key={id}>
+        <Link to={link}>{title}</Link>: {getProjectSolution(id, title)}
+      </li>
+    ));
+  };
+  const { username = '' } = props.user;
+  return (
+    <div>
+      As part of this certification, {props.name} built the following projects
+      and got all automated test suites to pass:
+      <Spacer />
+      {renderCertification(props.certName)}
+      <Spacer />
+      {solutionState.isOpen ? (
+        <Modal
+          aria-labelledby='solution-viewer-modal-title'
+          bsSize='large'
+          onHide={handleSolutionModalHide}
+          show={solutionState.isOpen}
+        >
+          <Modal.Header className='this-one?' closeButton={true}>
+            <Modal.Title id='solution-viewer-modal-title'>
+              Solution for {solutionState.projectTitle}
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <SolutionViewer
+              files={solutionState.files}
+              solution={solutionState.solution}
+            />
+          </Modal.Body>
+          <Modal.Footer>
+            <Button onClick={handleSolutionModalHide}>Close</Button>
+          </Modal.Footer>
+        </Modal>
```

Comment:
This modal is the same as the one in Certification.js, right?  If so, I think it's complicated enough to put in a separate component.  Also, we can drop `className='this-one?'` since it's never referenced.

Reply:
@ojeytonwilliams Sure, that makes sense. Where (directory) would `ProjectModal.js`(?) go?

Reply:
`client/src/components/SolutionViewerModal/index.js` makes the most sense to me.